### PR TITLE
chore(compass-aggregations): Replace react-dnd with react-sortable-hoc COMPASS-5655

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch is a brute force fix for COMPASS-5655 that addresses the issue by completely replacing the library. For some reason react-dnd backend singleton setup method started being called on every mount even though we create this backend only once, the root cause is somewhere deep in the library internals and we are 5 major versions behind, so there is no easy access to docs, searching through issues doesn't bring any relevant results, and updating the library would take more or less the same effort as replacing with something we already use in another place in Compass.

This does change how this whole thing works due to differences in how libraries work, so I wanted to highlight those:

- After the change the whole stage is dragged around, not the placeholder
- Because of that there is a noticeable lag in some cases when dropping an expanded stage because `react-ace` has to remount editors (and this always takes a decent amount of time)

|Before|After|
|---|---|
|Kapture 2022-04-12 at 13 55 11|Kapture 2022-04-12 at 13 47 23|

At the same time it's still better than a completely broken aggregation screen, so maybe it's a good enough compromise for now

### CODE REVIEW

File: packages/compass-aggregations/src/components/stage/stage.jsx [hunk 1]
```diff
@@ -135,9 +140,9 @@ class Stage extends Component {
           right: <ResizeHandle />,
         }}
       >
-        <StageEditorToolbar
+        <DragHandleToolbar
           allowWrites={this.props.allowWrites}
-          connectDragSource={this.props.connectDragSource}
+          // connectDragSource={this.props.connectDragSource}
```
Comment (addaleax) [inline packages/compass-aggregations/src/components/stage/stage.jsx:145]: Should these be removed entirely?

File: packages/compass-aggregations/src/components/stage-editor-toolbar/stage-editor-toolbar.jsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ class StageEditorToolbar extends PureComponent {
   static displayName = 'StageEditorToolbar';
   static propTypes = {
     allowWrites: PropTypes.bool.isRequired,
-    connectDragSource: PropTypes.func.isRequired,
+    // connectDragSource: PropTypes.func.isRequired,
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/components/stage-editor-toolbar/stage-editor-toolbar.jsx:20]:
```suggestion
```

File: packages/compass-aggregations/src/components/stage-editor-toolbar/stage-editor-toolbar.spec.jsx [hunk 3]
```diff
@@ -40,7 +40,7 @@ describe('StageEditorToolbar [Component]', function() {
         isReadonly={false}
         isTimeSeries={false}
         sourceName={null}
-        connectDragSource={(reactNode) => reactNode}
+        // connectDragSource={(reactNode) => reactNode}
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/components/stage-editor-toolbar/stage-editor-toolbar.spec.jsx:43]:
```suggestion
```

File: packages/compass-aggregations/src/components/stage/stage.jsx [hunk 4]
```diff
@@ -43,7 +48,7 @@ class Stage extends Component {
     isTimeSeries: PropTypes.bool.isRequired,
     isReadonly: PropTypes.bool.isRequired,
     sourceName: PropTypes.string,
-    connectDragSource: PropTypes.func.isRequired,
+    // connectDragSource: PropTypes.func.isRequired,
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/components/stage/stage.jsx:51]:
```suggestion
```

File: packages/compass-aggregations/src/components/stage/stage.jsx (hunk 1 above)
Comment (gribnoysup) [inline packages/compass-aggregations/src/components/stage/stage.jsx:145]:
```suggestion
```

**Review by mcasimir — APPROVED** [review]

**Comment by gribnoysup** [comment]
There were some weird CI failures in an unrelated test cases that were not reproducing locally at all, rebasing in hopes that this is just an evergreen issue that will go away

**Pull request merged**